tensorboard/backend/security_validator_test.py

```diff
@@ -0,0 +1,193 @@
+    self.make_request_and_maybe_assert_warn(
+        create_headers(content_type="application/json")
+    )
+
+    self.make_request_and_maybe_assert_warn(
+        create_headers(content_type=""),
+        "Content-Type is required on a Response"
+    )
+
+  def test_validate_x_content_type_options(self):
+    self.make_request_and_maybe_assert_warn(
+        create_headers(x_content_type_options="nosniff")
+    )
+
+    self.make_request_and_maybe_assert_warn(
+        create_headers(x_content_type_options=""),
+        'X-Content-Type-Options is required to be "nosniff"'
+    )
+
+  def test_validate_csp_text_html(self):
+    self.make_request_and_maybe_assert_warn(
+        create_headers(
+            content_type="text/html; charset=UTF-8",
+            content_secrurity_policy=(
+                "default-src 'self';script-src https://google.com;"
+            )
+        ),
+    )
+
+    self.make_request_and_maybe_assert_warn(
+        create_headers(
+            content_type="text/html; charset=UTF-8",
+            content_secrurity_policy="",
+        ),
+        "Requires default-src for Content-Security-Policy",
+    )
+
+    self.make_request_and_maybe_assert_warn(
+        create_headers(
+            content_type="text/html; charset=UTF-8",
+            content_secrurity_policy="default-src *"
+        ),
+        "Illegal Content-Security-Policy for default-src: *",
+    )
+
+    self.make_request_and_maybe_assert_warn(
+        create_headers(
+            content_type="text/html; charset=UTF-8",
+            content_secrurity_policy="default-src 'self';script-src *"
+        ),
+        "Illegal Content-Security-Policy for script-src: *",
+    )
+
+    self.make_request_and_maybe_assert_warn(
+        create_headers(
+            content_type="text/html; charset=UTF-8",
+            content_secrurity_policy=(
+                "script-src * 'sha256-foo' 'nonce-bar';"
+                "style-src http://google.com;object-src *;"
+                "img-src 'unsafe-inline';default-src 'self';"
+                "script-src *       'strict-dynamic'"
+            )
+        ), "\n".join(
+            [
+                "Illegal Content-Security-Policy for script-src: *",
+                "Illegal Content-Security-Policy for script-src: 'nonce-bar'",
+                "Illegal Content-Security-Policy for style-src: http://google.com",
+                "Illegal Content-Security-Policy for object-src: *",
+                "Illegal Content-Security-Policy for img-src: 'unsafe-inline'",
+                "Illegal Content-Security-Policy for script-src: *",
+                "Illegal Content-Security-Policy for script-src: 'strict-dynamic'",
+            ]
+        )
+    )
+
+    headers = create_headers(
+        content_type="text/html; charset=UTF-8",
+        content_secrurity_policy=(
+            "script-src * 'sha256-foo';"
+            "style-src http://google.com"
+        )
+    )
+    headers.add(
```

Comment:
If the point here is to test multiple CSP headers, IMO would be better to have a separate test method for this, both since it's testing a slightly different code path and since this test method is already quite long.